client/allochealth/tracker.go

```diff
@@ -502,30 +503,9 @@ OUTER:
 		// Detect if all the checks are passing
 		passed := true
 
-	CHECKS:
-		for _, treg := range allocReg.Tasks {
-			for _, sreg := range treg.Services {
-				for _, check := range sreg.Checks {
-					onUpdate := sreg.CheckOnUpdate[check.CheckID]
-					switch check.Status {
-					case api.HealthPassing:
-						continue
-					case api.HealthWarning:
-						if onUpdate == structs.OnUpdateIgnoreWarn || onUpdate == structs.OnUpdateIgnore {
-							continue
-						}
-					case api.HealthCritical:
-						if onUpdate == structs.OnUpdateIgnore {
-							continue
-						}
-					default:
-					}
-
-					passed = false
-					t.setCheckHealth(false)
-					break CHECKS
-				}
-			}
+		// scan for missing or unhealthy consul checks
+		if !evaluateConsulChecks(t.tg, allocReg) {
+			passed = false
```

Comment:
Is it meaningful that `t.setCheckHealth(false)` is no longer called alongside this bool being set?  I see that it is called elsewhere, but haven't chased down the potential implications of its removal here.

Reply:
Ah good catch, and I believe this is important in the edge case where

1. checks become healthy
2. tasks become healthy
3. start minimum healthy time
4. checks become unhealthy
5. minimum healthy time ends <- we never updated check status to unhealthy, so we'll incorrectly report they are healthy

Reply:
Added this back in, and also added `TestTracker_ConsulChecks_HealthToUnhealthy` to cover the case where health checks transition from healthy to unhealthy during the minimum health period.